Fix(org-stats) : NoProjects -> NoProjects("No projects available")

maxiuyuan opened the pull request:
changed NoProjects into NoProjects("No projects available") to give more context to the exception raised.

https://getsentry.atlassian.net/browse/ER-713

evanpurkhiser committed:
chore(prettier): Upgrade to 2.3.x (#26820)

BYK committed:
ci: Remove some checks from release branches (#26825)

wedamija committed:
feat(semver): Support equality match for semver filters (#26797)

This adds support for semver queries like `sentry.semver:1.2.4`, which makes an exact match.

Zylphrex committed:
feat(performance): Improve how errors are shown within a span (#26794)

The existing chevron style expand/collapse is difficult to find. This changes
the UI to use a show more/less button and shows a preview of the first 5 errors
within the span.

wedamija committed:
feat(semver): Add support for wildcard matching (#26808)

This adds support for wildcard matching when performing a semver search. This can look like:
 - `<key>:1.2.*`
 - `<key>:sentry@1.2.*`
 - `<key>:1.2.X`
 - `<key>:1.*`
 - etc

I've split the parsing logic out from `filter_by_semver`. I want to take this a step further and
have `filter_by_semver` not be aware of parsing at all, and just pass a `SemverFilter` to it
directly, but will handle that refactor separately.

iProgramStuff committed:
feat(workflow): Release Monitor (#26568)

taylangocmen committed:
fix(releases): Put pack release health cta for non mobile projects (#26837)

scttcper committed:
feat: Add semver autocomplete to issues search (#26803)

edwardgou-sentry committed:
feat(performance): transaction events tab sorting and column resizing (#26689)

Improved Transaction Summary All Events tab table
* Added column resizing
* Added column sorting
* Added sorting popper for operations column

scttcper committed:
fix(workflow): Adjust issue alert rule time period copy (#26839)

shruthilayaj committed:
fix(performance-setting): Add Project Permission Alert (#26827)

Add project permission alert on the projects performance
settings page.

markstory committed:
ref(ui) Remove createReactClass from withRelease (#26843)

Also do some cleanup on tests that I missed when withCommiters was
updated.

scttcper committed:
build(ui): Enable incremental builds for fork-ts-checker (#26842)

doralchan committed:
updated copy (#26845)

dashed committed:
ref(spans): Refactor span waterfall graph construction to mobx (#26746)


Co-authored-by: Tony Xiao <txiao@sentry.io>

mgaeta committed:
feat(notifications): Allow TEAM scope (#26756)

billyvg committed:
build(webpack): Change webpack to have smaller chunks and make all entries async (#25744)

Follow up to #25566 (and the revert #25681) - there was an issue where `jQuery` (as well as `SentryApp` global) is no longer available before the `DOMContentLoaded` event because our js bundles are loaded async with this PR. See https://github.com/getsentry/sentry/pull/25731, https://github.com/getsentry/sentry/pull/25728 for fixes

## Breaking changes
See [this forum post](https://forum.sentry.io/t/breaking-frontend-changes-for-custom-plugins/14184) for more information, but this has breaking changes for custom plugins (if you are hosting Sentry yourself).

### Original PR description
This changes our entrypoints to dynamically import the main bundles (i.e. the previous entrypoints). The goal here is to make the entrypoints as minimal as possible as we may need these to be uncached for splitting off frontend deploys.

The other change is removing the vendor chunk that we had, and instead use webpack defaults and let it decide how to chunk our build. This results in the client needing to request more + smaller chunks (instead of fewer + larger chunks). This costs a bit of network overhead, but a trade-off here is that the smaller chunks should have longer cache lives.

I'd like to push this out and evaluate how it affects our web vitals in prod and then make a decision on how we'd like to proceed.

Here's a light benchmarking of the perf changes, I used Chrome's dev tools to throttle a mix of the CPU and network bandwidth:

(tbh `DOMContentLoaded` doesn't seem like a very practical measurement here, but it's what was readily available in CDT)
![image](https://user-images.githubusercontent.com/79684/116126971-1e556300-a67c-11eb-948e-69e07ac3078b.png)

(private spreadsheet: https://docs.google.com/spreadsheets/d/1EN2jD0fqAJCASaAL1hAEBZJScTutmCZgCvDooamB3u4/)


### Locales
We were also previously loading locales for moment, but never setting the locale with `moment.locale()`, this PR addresses this as well as simplifying the webpack logic around locales (and instead pushing them into the application code).

scttcper committed:
build(ui): Add viewport tag to dev ui (#26855)

ceorourke committed:
feat(notification platform): Add slash command for team linking (#26750)

* feat(notification platform): Add slash command for team linking

taylangocmen committed:
fix(releases): Sync crash free users/sessions sort with display (#26853)

roggenkemper committed:
feat(growth) : Adds React Native events to Demo Mode  (#26790)

This PR adds 2 React Native events to Demo Mode. Previously, there was a React Native organization created but it was not populated with events.

priscilawebdev committed:
fix(grouping): Fix pagination caption description (#26835)

priscilawebdev committed:
fix(app-store-connect): Fix closeModal + refresh on save (#26828)

jjbayer committed:
feat(releases): List archived artifacts in files endpoint (#26351)

priscilawebdev committed:
ref(invite-members-modal): Update permission description (#26862)

priscilawebdev committed:
ref(feature-flag): Activate app-store-connect (#26861)

priscilawebdev committed:
fix(invite-members-modal): Remove repeated word (#26866)

jan-auer committed:
ref(settings): Sort Relays by last seen date (#26868)

k-fish committed:
fix(perf): Small UI tweaks on tags page (#26751)

Clean up truncation for tag values in the sidebar, fix the chart title background overlapping the graph, and have the other tags header conditionally removed based on if any tags exist.

Zylphrex committed:
fix(performance): Escape tag values when adding to query (#26844)

Tag values that are added to the query through the UI may contain special
characters like asterisks and slashes that need to be escaped. This ensures that
when adding them to the query, they are correctly escaped.

wmak committed:
feat(arithmetic): Add the ability to chart equations (#26758)

- This adds support for equations in the discover charts
- Added truncation to the y-axis labels if they go over 60 characters,
  this is cause we're changing from the `equation[0]` aliases to
  truncating equations if they get to long instead, will update the
  column headers in a different PR

scttcper committed:
fix(workflow): Use loading indicator without negative margin (#26849)

maxiuyuan committed:
inital for pairing

Zylphrex committed:
fix(lint): Broken contextData lint (#26871)

Zylphrex committed:
fix(discover): Do not replace on null value (#26872)

Not certain why but there can be nulls here. Adding a check to make sure it
doesn't error.

Commit:
feat(db): Add App Store Connect table (#26864)

For App Store Connect support we need a table of the builds that we
have downloaded.  This adds the database migration.

mgaeta committed:
types(python): Feature Flag types (#26672)

wedamija committed:
fix(semver): Fix bug in semver snuba related searches when we find no related releases. (#26841)

We'd get an error if a semver search returned no versions when making a snuba search (issues,
discover, etc). To work around this, if we find no versions then just insert a dummy release.

Alternatively, we could just raise an `InvalidSearchQuery("No releases found for this query")`.
This stops us making a pointless query to snuba, maybe that's preferable?

evanh committed:
ref(snql) Move more referrers to SnQL (#26840)

Zylphrex committed:
Revert "types(python): Feature Flag types (#26672)" (#26874)

This reverts commit e11b3c07c7486fd7c9f43406fc871da9fb1f81f7.

iProgramStuff committed:
changing some copy (#26875)

maxiuyuan committed:
for debugging

maxiuyuan committed:
test for er-713

maxiuyuan committed:
cleaning up spaces

maxiuyuan committed:
spacing

RyanSkonnord commented on tests/snuba/api/endpoints/test_organization_stats_v2.py:
```diff
@@ -33,7 +34,9 @@ def setUp(self):
         self.user2 = self.create_user(is_superuser=False)
+        self.user3 = self.create_user(is_superuser=False)
         self.create_member(user=self.user2, organization=self.organization, role="member", teams=[])
+        self.create_member(user=self.user3, organization=self.org3, role="member", teams=[])
```
Are these new entities still needed, or can we do the same test with the existing ones?

maxiuyuan replied:
I don't believe so, let me test it out

maxiuyuan replied:
after testing it, org3 is needed since the other orgs have projects within but i don't need user3 and could just set user2 as a member of org3.

maxiuyuan committed:
refactor to get rid of extra user

wedamija committed:
refs(semver): Move semver parsing code out of `ReleaseQuerySet` (#26810)

The semver parsing code isn't really a concern of the `Release` model or any related
managers/querysets. This pr extracts it out to the search filter layer, and passes in a
`SemverFilter` to `filter_by_semver` instead.

wedamija commented:
👋
